### DISCUSSION TITLE
How to identify whether a column referencing problem is caused by DBeaver or by the driver

### DISCUSSION THREAD

**Discussion by nchammas** [dbeaver/dbeaver#14504] [Q&A]
I am experiencing a weird issue. I am not sure if it's caused by DBeaver (Version 21.2.4.202111011351) or by the driver I'm using. I'm connecting to Spark on Databricks, using [Databricks's JDBC driver](https://databricks.com/spark/jdbc-drivers-download) (v2.6.19).

My query looks like this:

```sql
select 
    authors.pmid,
    authors.author_index,
    abc.npi,
    name_similarity,
    affiliation_similarity,
    authors.forename,
    ...
from
    some_table1 as authors
    inner join some_table2 as matches on ...
    inner join some_table3 as abc on ...
order by
    (name_similarity + affiliation_similarity) desc,
    authors.pmid,
    ...
;
```

The query works when I run it directly on Databricks, but when I run it via DBeaver it fails:

```
SQL Error [500051] [HY000]: [Simba][SparkJDBCDriver](500051) ERROR processing query/statement. 
Error Code: 0, SQL state: Error running query: org.apache.spark.sql.AnalysisException: cannot resolve 
'`matches.matchesname_similarity`' given input columns: [authorsaffiliation_cleaned, authorsauthor_index, 
authorsforename, authorslastname, authorspmid, abcaffiliated_name_clean, abcfirst_name, abclast_name, 
abcnpi, matchesaffiliation_similarity, matchesname_similarity]; line 1 pos 819;
```

The problem is with the `name_similarity` and `affiliation_similarity` columns in the `SELECT`. If I comment them out, the query works. Note that those same columns are referenced in the `ORDER BY` clause, so it's something specifically about the `SELECT` clause.

I notice that the columns listed by Spark after `cannot resolve` seem mangled, in that the table alias is concatenated with the column name without the `.` separator between them. I also see the alias repeated in some cases, like `matches.matchesname_similarity`.

This is all pretty strange. How do I know if this is caused by DBeaver, by the driver, or by something else?

**Comment by LonwoLonwo** [comment]
Hello @nchammas 
Sometimes DBeaver change queries before sending them to the database, but very rare.
You can check this in the [Query Manager ](https://github.com/dbeaver/dbeaver/wiki/Query-Manager)
